### DISCUSSION TITLE
Question about Transforms and code comments

### DISCUSSION THREAD

**Discussion by cfjedimaster** [11ty/eleventy#1967] [Q&A] [answered]
On this page, https://www.11ty.dev/docs/config/#transforms-example-minify-html-output, the code looks like so:

```
const htmlmin = require("html-minifier");

module.exports = function(eleventyConfig) {
  eleventyConfig.addTransform("htmlmin", function(content, outputPath) {
    // Eleventy 1.0+: use this.inputPath and this.outputPath instead
    if( outputPath && outputPath.endsWith(".html") ) {
      let minified = htmlmin.minify(content, {
        useShortDoctype: true,
        removeComments: true,
        collapseWhitespace: true
      });
      return minified;
    }

    return content;
  });
};
```

Notice specifically: ` // Eleventy 1.0+: use this.inputPath and this.outputPath instead` 

I'd like to know more about this. First (based on my discussion with @eleven_ty on Twitter) it sounds like we needed support for inputPath. That makes total sense to me!

What I'm not understanding is why support for inputPath is via `this`. In other words, I'd expect the callback form to support:

```
function(content, outputPath, inputPath)
```

So why are we using this.inputPath and this.outputPath instead? (From what I know the code sample will be updated to use that.)

**Comment by pdehaan** [comment, marked as answer]
I think maybe this: https://github.com/11ty/eleventy/issues/789; or more specifically https://github.com/11ty/eleventy/issues/789#issuecomment-629688643:

> The awkward part of this is that the linter and transform arguments are going to be reversed. Hmm.

Signatures from the current docs at https://www.11ty.dev/docs/config/#transforms

```js
eleventyConfig.addLinter("linter-name", function(content, inputPath, outputPath) {});
eleventyConfig.addTransform("transform-name", function(content, outputPath) {});
```

Also this comment further down in the thread https://github.com/11ty/eleventy/issues/789#issuecomment-719748095:

> Code is in for this (https://github.com/11ty/eleventy/commit/71f5943f1fa826a2b6d589cc1f49676d23b88afe). We’re using `this.inputPath` and `this.outputPath` to avoid the argument order consistency problem with linters. We’ll keep support for the existing arguments but transition away from them in the docs.

---



I also found https://github.com/11ty/eleventy/issues/1522 via the search. Not sure of the status of that and if it will change from `this.inputPath` to `this.page.inputPath`. 🤷

**Reply by cfjedimaster** [reply]
Ok... I guess. :)  It just feels weird to me to define a callback that ignores some of the arguments and recommends using this variables instead. So my callback will use:

* content
* this.inputPath
* this.outputPath

That just feels wrong to me. But - it's a stylistic opinion and I can get over it. ;)
